exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillStatementImpl.java

```diff
@@ -65,107 +65,52 @@ private void throwIfClosed() throws AlreadyClosedSqlException {
   // called ResultSet.)
 
   @Override
-  public DrillConnectionImpl getConnection() {
-    // Can't throw any SQLException because AvaticaConnection's getConnection() is
-    // missing "throws SQLException".
-    try {
-      throwIfClosed();
-    } catch (AlreadyClosedSqlException e) {
-      throw new RuntimeException(e.getMessage(), e);
-    }
+  public DrillConnectionImpl getConnection() throws SQLException {
+    checkOpen();
     return connection;
   }
 
-  // WORKAROUND:  Work around AvaticaStatement's code that wraps _any_ exception,
-  // even if SQLException, by unwrapping to get cause exception so caller can
-  // throw it directly if it's a SQLException:
-  // TODO:  Any ideas for a better name?
-  private SQLException unwrapIfExtra( final SQLException superMethodException ) {
```

Comment:
Exactly this method is not required since now we don't wrap `SQLException` exceptions into `SQLException`. Other methods, as I mentioned in the previous comment, methods in Avatica have a check for connection is open.